### ISSUE TITLE
Name commands

### ISSUE THREAD

**Issue by Noviny** [Noviny/changesets#16]
We want to make sure that the commands and the names of the commands make sense. I'm looking at making several changes from build-releases. As always, naming is hard. Feedback would be appreciated.

Current binary is `build-releases`, so commands are `build-releases [command]`.

New binary will be `changesets`, so commands are `changesets [command]`

Current commands are:

- `initialize`: get changeset default config set up, and create the `.changeset` dir
- `changeset`: add a new changeset
- `version`: consume changesets and update package versions
- `publish`: publish any packages that need it (effectively just bolt publish)

### Simple changes

- `initialize` to `init`

Reasoning: `init` is more standard.

- `changeset` to `add`

Since the base command is now `changeset`, add is now used to generate a new one

### Annoying changes

`version` and `publish` were originall introduced as names for bolt commands, which would mean they can mirror npm (and yarn) command names, since the global bolt will intercept them, so running `bolt version` or `bolt publish` would always be fine.

Changesets are not set up to be installed globally so you tend to create a package script to make running them easier. The most obvious choice is to include a command of the same name. For example:

```
"changeset": "build-releases changeset",
"version": "build-releases version",
"publish": "build-releases publish",
```

If you run `yarn version` however, it will run yarn's version command before looking for package scripts, which is a massive nonbo. So I want to rename both of these.

I want to rename `publish` to `release`, which I think is fairly obvious.

`version` is giving me grief.

Leading candidate is `consume`, which is very logical once you understand changesets, but is non-obvious when learning the tool in the way that `version` was. `update` makes it unclear if it will update a changeset or packages. `update-packages` might be the best, but doesn't sound good for single-repo use of changesets, and is still not amazingly clear.

So tentative new name suggestions are:

- `initialize` => `init`
- `changeset` => `add`
- `version` => `consume`
- `publish` => `release`

**Comment by Noviny** [comment]
Bad other option is encourage people to globally install it and then overwrite npm/yarn namespaces

**Comment by jesstelford** [comment]
Possible options off the top of my head for `version`:

- `changesets bump`
- `changesets bump-versions`
- `changesets process`
- `changesets run`
- `changesets commit`
- `changesets build`
- `changesets compile`
- `changesets prepare`
- `changesets prepare-release`
- `changesets release prepare`
- `changesets pre-release`

**Comment by jesstelford** [comment]
All the other renamings sound great 👍

**Comment by mitchellhamilton** [comment]
>  changeset to add

This makes complete sense to me on the surface if I was running `yarn changeset add` but having package scripts is generally what people end up doing. So with `add`, what would be the encouraged thing to with npm scripts, would you have a script called `changeset:add`?

What if the `changeset add` command existed but just running `changeset` would be equivalent to it, similar to how just running `yarn` is equivalent to `yarn install`. I think that it'd be nice since the majority of people interacting with changesets will only be adding a changeset so IMO, we should optimize for that case.

**Comment by Noviny** [comment]
Good point on optimising for how people will actually be using these (which I got halfway to). Agree that calling `changeset` without a command can just run add, even if there's also an add script.

```
"changeset": "changesets",
"consume-changesets": "changesets consume",
"release": "changesets release"
```

Alt take would be:

```
"changeset": "changesets",
"changeset:consume": "changesets consume",
"changeset:release": "changesets release"
```

but I feel people will shortcut the second to (or, I know I would)

Other point @mitchellhamilton accidentally brings up, is the base command `changeset` or `changesets`. I guess singular is more common for binaries?

**Comment by Noviny** [comment]
Released as of `@changesets/cli 1.0.1`
